library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationProvider.java

```diff
@@ -106,94 +114,97 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     @Override
-    public void onConnected(Bundle bundle) {
-        LogUtils.logI("GoogleApiClient is connected.");
-        boolean locationIsAlreadyAvailable = false;
+    public void onConnected() {
```

Comment:
Too many logic change here, the only change you'd need is as following

```
if (getConfiguration().keepTracking() || !locationIsAlreadyAvailable
              || waitingForConnectionToRequestLocationUpdate) {
            waitingForConnectionToRequestLocationUpdate(false);
            locationRequired();
```

to

```
if (getConfiguration().keepTracking() || !locationIsAlreadyAvailable) {
            locationRequired();
```

Reply:
This code moved to requestLocation(), because checkLastKnowLocation() is task now, we cannot get locationIsAlreadyAvailable value in this function.

Same logic is saved, I think.